model/host/host.go

```diff
@@ -3033,53 +3045,60 @@ func GetPaginatedRunningHosts(hostID, distroID, currentTaskID string, statuses [
 
 	hasFilters := false
 
-	if len(hostID) > 0 {
+	if len(opts.HostID) > 0 {
 		hasFilters = true
 
 		runningHostsPipeline = append(runningHostsPipeline, bson.M{
 			"$match": bson.M{
-				IdKey: hostID,
+				"$or": []bson.M{
+					{
+						IdKey: bson.M{"$regex": opts.HostID, "$options": "i"},
```

Comment:
Is there a user reason we're making this regex or did it just seem like a possible improvement? This is worse performance-wise and I don't think is necessary for this -- when it comes to searching by host name or DNS name, users (probably always) copy paste the thing they need to search for and paste it.

Reply:
I wanted to comment on this as well. But I figured a regex search might allow users to perform searches like the below search to match all hosts that match a certain pattern. For example, matching on all Macos hosts or Zseries hosts that have these static ids
<img width="1196" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/4605522/0b16853d-c8d7-4019-a886-f6f0b66cf29a">

Reply:
Geoff was using the legacy hosts page to do regex search, so I just figured that's what he would want on new UI too. I can remove it though

Reply:
Does the legacy host page support regex ? I'm inclined to remove it as unsupported myself; presumably you could filter on distro for other things but if we think there's a use case we can hold it.

Reply:
the legacy page doesn't support regex but it supports includes (like `hostName.includes("my-filter")`) — but other than the examples Mohamed mentioned, it’s true that it might not be that useful, so I removed it for now 🫨